### ISSUE TITLE
Handle abnormal capitalization in titleCase

### ISSUE THREAD

**Issue by platers** [platers/obsidian-linter#41]
Terms such as acronyms are not handled properly. It seems hard to support every edge case.
The rule could only apply to lowercase words, and ignore other words. I think this should be an option.

**Comment by chrisgrieser** [comment]
I agree. Trying to come up with a more generalized solution would also be more language-independent. The short words to ignore, for example, are all in English right now.  However, I would define the "exception rules" a bit differently. 

I imagine a benefit of the rule to some could also be the automatic correction of typos like `# FOobar` which will be corrected as well. stipulating that the rule should only apply to lowercase words would remove with nice side-benefit.

Here is what i would suggest as sensitive title-case-exception-rules
1. all words with 3 or less characters are lowercased (except when at the beginning)
2. all words with 4 or more characters that are combinations of two words from 1) are also lowercased (except when at the beginning). This ensures words like "into" stay lowercase for consistency.
3. all words that are *completely* uppercased will stay so to ensure, as it is assumed that they are acronyms.

This leaves some edge cases (e.g. "with") and words with special casing (e.g. "iPhone"). The number of those cases should however be quite small, so it could make sense to keep a list of such words, like the titlecase rule does right now.

**Comment by platers** [comment]
I think English is too complicated to handle all the edge cases. Your logic would skip words like cat, and there are countless other cases. 

I found this popular [implementation](https://github.com/blakeembrey/change-case/blob/master/packages/title-case/src/index.ts) which looks good to me.

Running Linter should never make your document worse, since its run so often.

**Comment by chrisgrieser** [comment]
You are right, I always forget how many short words English has. However, this leaves the problem how to titlecase correctly in languages other than English. Is there maybe some solution like "change-case" available that also works on most common languages?

**Comment by platers** [comment]
Not afaik. Maybe implementing a way to turn off rules in a file would alleviate this issue. #46

**Comment by trungng2006** [comment]
I also ran into this problem.
I think there are 2 ways to resolve this, the plugin can either let the user create a list of word that:
1. ignore the heading titleCase
OR
2. Alway use AllCaps

Cause the people who wrote their note know the acronym they want to use. I think letting the user define their own list can resolve most of the edge case.
